src/java.base/share/classes/java/io/SimpleIO.java

```diff
@@ -0,0 +1,1031 @@
+    }
+
+    /**
+     * Print list of lines to output stream. Each line will be printed and
+     * followed by a line terminator.
+     * For example:
+     * {@snippet lang="java":
+     * var fruit = List.of("apple", "pear", "orange");
+     * printLines(fruit); // @highlight substring="printLines"
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * apple
+     * pear
+     * orange
+     *
+     * }
+     *
+     * @param lines list of strings
+     * @throws NullPointerException if lines is null
+     */
+    public static void printLines(List<String> lines) {
+        Objects.requireNonNull(lines, "lines must not be null");
+        lines.forEach(System.out::println);
+    }
+
+    /**
+     * The contents of the file {@code filename} are read as a string.
+     * For example:
+     * {@snippet lang="java":
+     * var text = """
+     *    "Hope" is the thing with feathers
+     *    That perches in the soul
+     *    And sings the tune without the words
+     *    And never stops - at all
+     *    """;
+     * write("data.txt", text);
+     * var data = read("data.txt"); // @highlight substring="read"
+     * print(data);
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * "Hope" is the thing with feathers
+     * That perches in the soul
+     * And sings the tune without the words
+     * And never stops - at all
+     *
+     * }
+     *
+     * @param filename  string representing the name or path of the file to be read
+     * @return Content read from a file
+     * @throws UncheckedIOException wrapping an IOException if an io error occurs.
+     *
+     * @implSpec Line terminators are normalized to '\n'.
+     */
+    public static String read(String filename) {
+        Objects.requireNonNull(filename, "filename must not be null");
+        return read(Path.of(filename));
+    }
+
+    /**
+     * The contents of the file at {@code path} are read as a string.
+     * For example:
+     * {@snippet lang="java":
+     * var text = """
+     *    "Hope" is the thing with feathers
+     *    That perches in the soul
+     *    And sings the tune without the words
+     *    And never stops - at all
+     *    """;
+     * var path = Path.of("data.txt");
+     * write(path, text);
+     * var data = read(path); // @highlight substring="read"
+     * print(data);
+     * }
+     * will print on the output stream as:
+     * {@snippet lang="text":
+     * "Hope" is the thing with feathers
+     * That perches in the soul
+     * And sings the tune without the words
+     * And never stops - at all
+     *
+     * }
+     *
+     * @param path  path of file to be read
+     * @return Content read from a file
+     * @throws UncheckedIOException wrapping an IOException if an io error occurs.
+     *
+     * @implSpec Line terminators are normalized to '\n'.
+     */
+    public static String read(Path path) {
```

Comment:
Unless I missed something, this would be the first class in the `java.io` package to have a `java.nio.file.Path` parameter. The only other use of `Path` is by `File::toPath`. Should `Path` be replaced with `File` for parameters?

Reply:
> Unless I missed something, this would be the first class in the `java.io` package to have a `java.nio.file.Path` parameter. The only other use of `Path` is by `File::toPath`. Should `Path` be replaced with `File` for parameters?

I think we should expect new APIs to use Path rather than File but for this API, there is an argument that all the methods should take a path string rather than having two methods for each op.

Reply:
Also, the `SimpleIO::{read,write}*(Path...)` methods do not add any value over their `java.nio.file.Files` counterparts.